react/components/ProductSummary.js

```diff
@@ -17,11 +19,34 @@ import { useCssHandles } from 'vtex.css-handles'
 const PRODUCT_SUMMARY_MAX_WIDTH = 300
 const CSS_HANDLES = ['container', 'containerNormal', 'element', 'clearLink']
 
-const ProductSummaryCustom = ({ product, actionOnClick, children, containerRef }) => {
+const ProductSummaryCustom = ({
+  product,
+  actionOnClick,
+  children,
+  containerRef,
+}) => {
   const { isLoading, isHovering, selectedItem, query } = useProductSummary()
   const dispatch = useProductSummaryDispatch()
   const handles = useCssHandles(CSS_HANDLES)
 
+  // Use ProductListContext to send pixel events
+  const { useProductListDispatch } = ProductListContext
+  const productListDispatch = useProductListDispatch()
+  const [inViewRef, inView] = useInView({
+    // Triggers the event when the element is 75% visible
+    threshold: 0.75,
+    triggerOnce: true,
+  })
+  useEffect(() => {
+    if (inView) {
+      productListDispatch &&
```

Comment:
it would be nice to have a comment here saying why you need to check if this is truthy

Reply:
Sure!